firmware/main.c

```diff
@@ -692,131 +689,128 @@ static void cmd_motor_rotary(BaseSequentialStream *chp, int argc, char *argv[])
 	palSetPadMode(pinPorts[pinMode2].gpio, pinPorts[pinMode2].pin, PAL_MODE_OUTPUT_PUSHPULL);
 
 	// set mode pins
-	chprintf(chp, "Mode %d selected\r\n", mode);
-	int gpioLevelMode = mode-1;
-	if(gpioLevelMode & 1)
+	if(mode == ROTATE) {
 		palSetPad(pinPorts[pinMode1].gpio, pinPorts[pinMode1].pin);
-	else
-		palClearPad(pinPorts[pinMode1].gpio, pinPorts[pinMode1].pin);
-	if(gpioLevelMode & 2)
-		palSetPad(pinPorts[pinMode2].gpio, pinPorts[pinMode2].pin);
-	else
 		palClearPad(pinPorts[pinMode2].gpio, pinPorts[pinMode2].pin);
+		}
+	else if(mode == HOME) {
+		palSetPad(pinPorts[pinMode1].gpio, pinPorts[pinMode1].pin);
+		palSetPad(pinPorts[pinMode2].gpio, pinPorts[pinMode2].pin);
+		}
+
+
+	char direction_literal[25];
+	float rotational_angle;
 
 	switch(mode) {
-		case 1: // Clock/direction mode CW
-			// !fall through!
-		case 2: // Clock/direction mode CCW
-			// !fall through from above!
-			chprintf(chp, "Pin to use: %s, Pulses to produce %d\r\n",
+		// 1: rotate
+		case 1:
+			chprintf(chp, "Selected mode: rotate\n\r\n");
+			chprintf(chp, "Pin to use: %s, Pulses to produce %d\n\r\n",
 					pinPorts[pinPulses].pinNrString,
 					geared_pulses);
 			chThdSleepMilliseconds(200);
 			pulses_equidistant(pinPorts[pinPulses].gpio, pinPorts[pinPulses].pin, geared_pulses, false, 2);
-			chprintf(chp, "Sent %d Pulses %d direction\r\n", geared_pulses, direction);
-			if(1 == direction)
-				palClearPad(pinPorts[pinDirection].gpio, pinPorts[pinDirection].pin);
+			if(direction == CW)
+				sprintf(direction_literal, "clockwise (CW)");
+			else if(direction == CCW)
+				sprintf(direction_literal, "counterclockwise (CCW)");
+			// Each sent pulse will result in 1.8 degrees rotation
+			rotational_angle = pulses * 1.8;
+			chprintf(chp, "Number of pulses sent: %d\r\nDirection: %s\r\n", geared_pulses, direction_literal);
+			chprintf(chp, "Angle of rotation: %.1f\r\n", rotational_angle);
+			// reset direction pin at the end of the operation to 0
+			palClearPad(pinPorts[pinDirection].gpio, pinPorts[pinDirection].pin);
 			break;
-
-		case 3: // Turn 45 degree CCW
-			// !fall through!
-		case 4: // Homing Mode
-			// !fall through from above!
+		// 2: home
+		case 2:
+			chprintf(chp, "Selected mode: home\r\n");
+			chprintf(chp, "\tRotary unit returns to reference position\r\n");
 			palSetPad(pinPorts[pinEnable].gpio, pinPorts[pinEnable].pin);
 			chThdSleepMilliseconds(200);
 			palClearPad(pinPorts[pinEnable].gpio, pinPorts[pinEnable].pin);
 			break;
-
 		default:
 			chprintf(chp, "Select one of the available modes\r\n");
 			break;
 	}
 	return;
 
+
 exit_with_usage:
-	chprintf(chp, "Usage: motor_rotary -m <mode> -p [pulses] -d [direction]\r\n"
-			"\tNumber of pulses to turn (1 pulse = 1,8 degrees the motor and 0,0375 degrees the rotary table)\r\n"
+	chprintf(chp, "Usage: motor_rotary <mode> <arguments>\r\n"
+			"\tNumber of pulses to turn "
```

Comment:
the helptext is a bit off now.
make sure its concise, clear and well structured. e.g.:
- generic calls (maybe multiple lines)
- description of generic calls
- description of each parameter

just take `random command --help` as an example here.